File: sqlitejava/src/main/java/com/codekul/sqlitejava/CarDao.java
package com.codekul.sqlitejava;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;
import android.database.Cursor;

import java.util.List;

/**
 * Created by aniruddha on 16/11/17.
 */

@Dao
public interface CarDao {

    @Insert
    void insert(Car car);

    @Query("select * from Car")
    List<Car> cars();

    @Query("select * from Car")
    Cursor allCars();
}

File: sqlite/src/main/java/com/codekul/sqlite/dao/CarDao.kt
package com.codekul.sqlite.dao

import android.arch.persistence.room.*
import com.codekul.sqlite.model.Car

/**
 * Created by aniruddha on 14/11/17.
 */
@Dao
interface CarDao {

    @Insert
    fun insert(car: Car)

    @Update
    fun update(car: Car)

    @Delete
    fun delete(id: Long)

    @Query("select * from Car")
    fun select(): List<Car>
}
